src/buf/into_buf.rs

```diff
@@ -1,6 +1,14 @@
 use super::{Buf};
 
-use std::io;
+#[cfg(feature = "use_alloc")]
+use alloc::string::String;
+#[cfg(feature = "use_alloc")]
+use alloc::vec::Vec;
+
+#[cfg(not(feature = "use_std"))]
+use buf::Cursor;
```

Comment:
Unfortunately, given that features must be additive, I don't think this conditionally vendoring cursor is something that can be done.

This, of course, means that `IntoBuf` can't really be implemented without `std` :(